File: 036/036.c
#include <stdio.h>

int is_palindrome(int number, int base)
{
    int result = 0, copy = number;
    
    while (copy > 0) {
        result = (result*base) + (copy % base);
        copy  /= base;
    }
    return result == number;
}

int main(void)
{
    int result = 0, count = 1;
    
    for (count; count < 1000000; ++count) {
        if (   is_palindrome(count, 2)
            && is_palindrome(count, 10)) {
            result += count;
        }
    }
    printf("%d\n", result);
}

File: 036/036.rb
class Fixnum
  def is_palindrome
    (self == self.reverse(2) and self == self.reverse(10))
  end
  def reverse(base)
    result,copy = 0, self
    while copy > 0
      result = (result*base) + (copy % base)
      copy  /= base
    end
    return result
  end
end

result = 0
1.upto(1000000) do |n|
    result += n.is_palindrome ? n : 0
end
puts result
